contracts/src/messaging/IMerkleTreeMessageBus.sol

```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity >=0.7.0 <0.9.0;
+
+import "./Structs.sol";
+
+interface IMerkleTreeMessageBus {
+    // This function is called to add a cross chain state root to the message bus.
+    function addStateRoot(bytes32 stateRoot, uint256 activationTime) external;
+    // This function disables a cross chain state root from the message bus. On challenge
+    function blockStateRoot(bytes32 stateRoot) external;
+    // This function verifies that a cross chain message is included in the state root.
+    // message - the message to verify
+    // proof - merkle tree proof for the said message against root
```

Comment:
Doesn't the proof already contain the root?

Reply:
it builds to a root, but then we need to compare it and check if it exists. It's just a requirement from how the OpenZeppelin library works.